### PR TITLE
Add tinycbor c++ library package

### PR DESCRIPTION
**Pull request by sdebionne**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sdebionne** [comment]
@conda-forge-admin, please rerender

**Comment by sdebionne** [comment]
@conda-forge/help-c-cpp, ready for review!

**Comment by sdebionne** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/tinycbor/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  # sha256 is the preferred checksum -- you can get it for a file with:
+  #  `openssl sha256 <file name>`.
+  # You may need the openssl package, available on conda-forge:
+  #  `conda install openssl -c conda-forge``
+  patches:
+    - 0001-Add-CMake-support.patch
+    - 0002-Fix-static_assert-compiler-support.patch
+
+build:
+  # Uncomment the following line if the package is pure Python and the recipe is exactly the same for all platforms.
```
Comment (carterbox) [inline recipes/tinycbor/meta.yaml:31]: Please remove all the explainer comments if you're done with them. The comments are still available in the reference recipe.

File: recipes/tinycbor/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  sha256: 512e2c9fce74f60ef9ed3af59161e905f9e19f30a52e433fc55f39f4c70d27e4
+  # sha256 is the preferred checksum -- you can get it for a file with:
+  #  `openssl sha256 <file name>`.
+  # You may need the openssl package, available on conda-forge:
+  #  `conda install openssl -c conda-forge``
+  patches:
+    - 0001-Add-CMake-support.patch
+    - 0002-Fix-static_assert-compiler-support.patch
+
+build:
```
Comment (carterbox) [inline recipes/tinycbor/meta.yaml:30]:
```suggestion
build:
  run_exports:
    - {{ pin_subpackage( name|lower, max_pin='x.x.x' ) }}
```
Please add a run export that chooses the appropriate API package version for this packages ABI compatability guarantees.

File: recipes/tinycbor/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (carterbox) [inline recipes/tinycbor/LICENSE:1]: You don't need to include a copy of the license? Isn't it included in the source archive?

File: recipes/tinycbor/0002-Fix-static_assert-compiler-support.patch [hunk 4]
```diff
@@ -0,0 +1,25 @@
+From 6639f6db4d0a0b88dbbd7c0eda6a2b1e08ae22d2 Mon Sep 17 00:00:00 2001
+From: Samuel Debionne <samuel.debionne@esrf.fr>
+Date: Wed, 23 Aug 2023 16:14:19 +0200
+Subject: [PATCH 2/2] Fix static_assert compiler support
+
+---
+ src/compilersupport_p.h | 2 +-
```
Comment (carterbox) [inline recipes/tinycbor/0002-Fix-static_assert-compiler-support.patch:7]: Have you opened a pull request to upstream this patch?

File: recipes/tinycbor/0001-Add-CMake-support.patch [hunk 5]
```diff
@@ -0,0 +1,209 @@
+From e0fcc34f29633e8d4213581ad1c40575f9d2a0f1 Mon Sep 17 00:00:00 2001
+From: Samuel Debionne <samuel.debionne@esrf.fr>
+Date: Wed, 23 Aug 2023 16:13:36 +0200
```
Comment (carterbox) [inline recipes/tinycbor/0001-Add-CMake-support.patch:3]: Have you opened a PR to upstream this patch? Does this CMake build system generate artifacts that are named the same as the ones generated by makefiles? For example, sometimes the DLL name on windows might have/not have the lib prefix.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/tinycbor/0002-Fix-static_assert-compiler-support.patch (hunk 4 above)
Comment (sdebionne) [reply recipes/tinycbor/0002-Fix-static_assert-compiler-support.patch:7]: I have just opened https://github.com/intel/tinycbor/pull/242

File: recipes/tinycbor/0001-Add-CMake-support.patch (hunk 5 above)
Comment (sdebionne) [reply recipes/tinycbor/0001-Add-CMake-support.patch:3]: I am not sure the original build system supports Windows, I'll need to check.

File: recipes/tinycbor/LICENSE (hunk 3 above)
Comment (sdebionne) [reply recipes/tinycbor/LICENSE:1]: Correct the LICENCE is provided in tarball.

**Comment by sdebionne** [comment]
Thank you for the through review (nice to get a review from a synchrotron colleague BTW). I applied the necessary changes.

**Comment by carterbox** [comment]
Sorry about the wait. Somehow the review-requested tag wasn't added, so I haven't been seing this PR in the list.

**Review by carterbox — APPROVED** [review]
LGTM if build passes. Just be aware that if the custom CMake script if not ABI compatible with upstream's build scripts, you may get complaints from users about compatability with other conda channels. Compatability with other channels is not something we support, but it's good to have artifacts which match the canonical ones.

**Pull request merged**